node/src/components/consensus/consensus_protocol.rs

```diff
@@ -85,6 +92,8 @@ pub(crate) enum ProtocolOutcome<I, C: Context> {
     NewEvidence(C::ValidatorId),
     /// Send evidence about the validator from an earlier era to the peer.
     SendEvidence(I, C::ValidatorId),
+    /// Send evidence about an equivocation to the event stream.
+    EquivocationEvent(Evidence<C>),
```

Comment:
We shouldn't leak Highway details; that breaks pluggability for consensus.

Reply:
`Evidence<C>` is sent to the `era_supervisor` and is then mapped onto an external type `EvidenceStream` to be sent to the event stream. Would that still classify as leaking details?

Reply:
Yes, `Evidence` is Highway-specific.

Reply:
@afck Maybe have the context `C` have an associated `C::Evidence` ?

Reply:
`Context` isn't implemented by the protocol, but… by all the rest! The `ClContext` specifies which types the Casper node is using, even though `Highway` could work with other contexts (and does, for faster and simpler tests).

Anyway, we decided to just push non-Highway-specific information on the event stream anyway; @darthsiroftardis will create a new PR that should be simpler.